src/core/ModelAnimator.ts

```diff
@@ -312,6 +338,62 @@ class ModelAnimator {
 
     this._fadePromises = [];
   }
+
+  private _updateRepeatMode() {
+    const view3D = this._view3D;
+    const actions = this._actions;
+    const repeatMode = view3D.animationRepeatMode;
+
+    if (repeatMode === ANIMATION_REPEAT_MODE.NONE) {
+      actions.forEach(action => {
+        action.clampWhenFinished = true;
+        action.loop = THREE.LoopOnce;
+      });
+    } else {
+      actions.forEach(action => {
+        action.clampWhenFinished = false;
+        action.loop = THREE.LoopRepeat;
+      });
+    }
+  }
+
+  private _onAnimationLoop = (evt: THREE.Event) => {
+    const view3D = this._view3D;
+    const actions = this._actions;
+    const clips = this._clips;
+    const index = actions.findIndex(action => action === evt.action);
+
+    if (view3D.animationRepeatMode === ANIMATION_REPEAT_MODE.ALL) {
+      const nextIndex = (index + 1) >= clips.length
+        ? 0
+        : index + 1;
+
+      this.play(nextIndex);
+    }
+
+    view3D.trigger(EVENTS.ANIMATION_LOOP, {
```

Comment:
I think it's a loop when all animations are finished and it goes back to the beginning.

"one": 1 => loop => 1 => loop => 1 => loop
"all": 1 => 2 => 3 => loop => 1 => 2 => 3 => loop

Reply:
Good point, I should change the event name to `animationChange` in that case.

Reply:
I rechecked the whole spec, and I think it would be better to maintain the name as `animationLoop` as it means a loop of a single animation.
If we change the name to `animationChange`, animation `play` also requires the event to be triggered.